src/ui/git/GitCredentialsWindow.cpp

```diff
@@ -13,18 +12,21 @@
 #include <LayoutBuilder.h>
 #include <TextControl.h>
 
+#include "GitCredentialsWindow.h"
+#include "GitRepository.h"
+
 
 #undef B_TRANSLATION_CONTEXT
 #define B_TRANSLATION_CONTEXT "GitCredentialsWindow"
 
-GitCredentialsWindow::GitCredentialsWindow(const char* username, const char* password)
+GitCredentialsWindow::GitCredentialsWindow(BString &username, BString &password)
 	:
 	BWindow(BRect(0, 0, 300, 150), B_TRANSLATE("Git - User Credentials"),
 			B_TITLED_WINDOW, B_NOT_RESIZABLE | B_NOT_ZOOMABLE
 			| B_AUTO_UPDATE_SIZE_LIMITS | B_NOT_CLOSABLE)
 {
-	fUsernameString = username;
-	fPasswordString = password;
+	fUsernameString = &username;
```

Comment:
Not sure why we are doing this ( using BString pointers).. to me there is the risk to have invalid memory access..

Reply:
What do you suggest? The BStrings are passed to the window by ref to retain their value after the window is closed. 
This piece is taken from TrackGit where it was even worse as plain char* are used. Is there a better way to do it?

Reply:
Why can't you use simple access methods like GetUsername or GetPassword? morehover 
the char username[39], password[128]; are uninitialized and converting to BString may lead to unexpected result as the size may be unknown/wrong.

Reply:
Haven't looked deeply, but can't we use a BMessage to return thise strings?

Reply:
Forget my last section of the previous message: I was looking at the wrong branch. No problem related to uninitialized variables. Anyway I'm for making access methods.

Reply:
That does not seem possible. The caller is a notify callback owned by libgit2 at that point in the life cycle. It waits for the login window to close with wait_for_thread() and retrieves the value of the two BStrings (username and password). It then passes the credentials to the main libgit2 function which calls the notify callback again and again if necessary until the credentials are correct or the user hits Cancel.
There is no easy way to use a BMessage or using accessor methods because the BWindow will be closed and invalid at that point.